client/my-sites/email/mailboxes/index.tsx

```diff
@@ -129,6 +147,15 @@ const MailboxesManagement = ( {
 		return <MailboxSelectionList domains={ nonWPCOMDomains } />;
 	}
 
+	if ( selectedSiteId && ! hasAtLeastOneMailbox( nonWPCOMDomains ) && newEmail ) {
```

Comment:
```suggestion
	if ( newEmail ) {
```
Two things strike me:
1. Checking `! hasAtLeastOneMailbox( nonWPCOMDomains )` is superfluous because the previous if statement already catches that.
2. We don't need to check for `selectedSiteId` here if we also do it in the conditional before `QuerySiteDomains`. We should pick one of those locations.